src/TransferList/TransferList.test.js

```diff
@@ -0,0 +1,98 @@
+import { render, screen, within } from "@testing-library/react";
+import React from "react";
+import TransferList from ".";
+import userEvent from "@testing-library/user-event";
+
+/**
+ * Test wrapper for TransferList
+ * Provides state for the selectedItems
+ */
+const SelectedItemsWithState = ({
+  items: itemsIn = ["Apples", "Pears", "Oranges"],
+  onChange,
+  selectedItem: selectedItemsIn = ["Apples"],
+  ...rest
+}) => {
+  const [selectedItems, setSelectedItems] = React.useState(selectedItemsIn);
+  const handleChange = selectedItem => {
+    setSelectedItems(selectedItems);
+    onChange && onChange(selectedItems);
+  };
+  return (
+    <TransferList
+      {...rest}
+      items={itemsIn}
+      onChange={handleChange}
+      selectedItems={selectedItems}
+    />
+  );
+};
+
+describe("TransferList", () => {
+  test("Search bar updates value", async () => {
+    const user = userEvent.setup();
+    const { container } = render(<SelectedItemsWithState />);
+
+    const inputBase = container.querySelector(".MuiInputBase-input");
+    await user.type(inputBase, "pears");
+    expect(inputBase.value).toBe("pears");
+  });
+
+  test("On click clearAll button should has text 'None Selected'", () => {
+    const user = userEvent.setup();
+    const onChange = jest.fn();
+    const { container } = render(
+      <SelectedItemsWithState onChange={onChange} selectedItem={[]} />
+    );
+
+    user.click(screen.queryByTestId("clear-all"));
+
+    expect(container.querySelector(".MuiTypography-body1").textContent).toBe(
```

Comment:
I think it would be good to also check onChange gets called with correct item (empty in this case).